src/pages/chainlink-functions/tutorials/api-use-secrets-offchain.md

```diff
@@ -0,0 +1,277 @@
+---
+layout: ../../../layouts/MainLayout.astro
+section: chainlinkFunctions
+date: Last Modified
+title: "Using Off-chain Secrets in Requests"
+setup: |
+  import ChainlinkFunctions from "@features/chainlink-functions/common/ChainlinkFunctions.astro"
+---
+
+This tutorial shows you how to leverage the Off-chain Secrets capability to share encrypted secrets off-chain via HTTP with a Decentralized Oracle Network.
+Off-chain secrets are encrypted, users decide where to host them (e.g., aws S3, google drive, or IPFS), and fetched by the Decentralized Oracle Network via HTTP(s).
+Using off-chain secrets has two main advantages:
+
+- Security: The encrypted secrets are never stored on-chain. Users choose where to store their encrypted secrets and pass the URLs to the encrypted secrets in their requests. The URLs are encrypted with the DON public key so that only an oracle node part of the DON can decrypt the URLs with the DON private key. Once the DON has fulfilled the request, a user may delete the URL to mitigate the risk of exposing their secrets if the DON's private key were ever to be leaked.
```

Comment:
```suggestion
- Security: The encrypted secrets are never stored on-chain. You choose where to store encrypted secrets and include the URLs in your requests. The secrets are encrypted by the DON's public key so that only an oracle node in the DON can decrypt them using the DON's private key. After the DON fulfills a request, you can delete the secrets file from the hosted URL. This mitigates the risk that your secrets are exposed if the DON's private key were ever to be leaked.
```